### PR TITLE
PIL-1215 Increase test coverage to meet Live Service requirements

### PR DESCRIPTION
**Pull request by JamesMMiller**
This PR adds test coverage to meet the requirement for Live Services, adds tests and removes unused code to meet the magic 90%.

Changes:
- Added comprehensive test coverage for RepaymentConnector using WireMock
  - Tests cover all expected response codes (201, 400, 409, 422, 500, 503)
  - Tests include both successful and error scenarios
  - Added edge case tests (empty/malformed responses) for completeness, though connector simply passes responses through to the repayment service which is pretty basic in of itself:
  ```
  def sendRepaymentsData(rePaymentData: RepaymentRequestDetail)(implicit hc: HeaderCarrier): Future[Done] =
    repaymentConnector.sendRepaymentDetails(rePaymentData).flatMap { response =>
      if (response.status == 201) {
        Future.successful(Done)
      } else {
        Future.failed(UnexpectedResponse)
      }
    }
  ```
- Removed unused SessionIdHelper utility class
- Added test coverage for LimitedLiabilityPartnership flow in SubscriptionService
  - Tests cover happy path scenarios for both create and read operations
- Increased coverage minimum requirement from 80% to 90% to meet Live Service standards

### CODE REVIEW

File: test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala [hunk 1]
```diff
@@ -516,9 +567,10 @@ trait ModelGenerators {
 
   val arbitraryWithIdRegDataFoLLP: Arbitrary[GrsResponse] = Arbitrary {
     for {
+      companyProfile                    <- arbitrary[CompanyProfile]
       partnershipEntityRegistrationData <- arbitrary[PartnershipEntityRegistrationData]
 
-    } yield GrsResponse(partnershipEntityRegistrationData = Some(partnershipEntityRegistrationData))
+    } yield GrsResponse(partnershipEntityRegistrationData = Some(partnershipEntityRegistrationData.copy(companyProfile = Some(companyProfile))))
```
Comment (ridow-hmrc) [inline test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala:573]:
```
partnershipEntityRegistrationData <- arbitrary[PartnershipEntityRegistrationData].suchThat(_.companyProfile.isDefined)
```

File: test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala [hunk 2]
```diff
@@ -89,6 +89,44 @@ class SubscriptionServiceSpec extends BaseSpec with Generators with ScalaCheckPr
...
+
+        forAll(arbitrary[String], Gen.option(arbitrary[String]), arbitraryWithIdUpeFmUserDataLLPUserAnswers.arbitrary) {
+          (upeSafeId, fmSafeId, userAnswers) =>
+            service.sendCreateSubscription(upeSafeId, fmSafeId, userAnswers).map { response =>
+              response.status mustBe OK
+            }
+        }
+      }
+
+      // "throw exception when company profile is missing" in {
```
Comment (ridow-hmrc) [inline test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala:112]: do we need this?

File: test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala (hunk 1 above)
Comment (JamesMMiller) [reply test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala:573]: It is neater but I've grown to dislike the suchTaht method as it just spawns another generator that filters down the output, I tend to prefer to ensure EVERY item I generate matches what I want

Comment (JamesMMiller) [reply test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala:573]: But for a simple case like this it's probably fine

File: test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala (hunk 2 above)
Comment (JamesMMiller) [reply test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala:112]: Ah yeah, probably.  apologies did this on friday and just put up the PR today so I kinda forgot about this - will update

File: test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala [hunk 3]
```diff
@@ -89,6 +90,50 @@ class SubscriptionServiceSpec extends BaseSpec with Generators with ScalaCheckPr
...
+          }
+        }
+      }
+
+      "throw exception when company profile is missing" in {
+        val llpWithoutCompanyProfile = arbitraryWithIdUpeFmUserDataLLP.arbitrary
+        val userAnswersGen           = userAnswersFromGenerators(Arbitrary(llpWithoutCompanyProfile))
+
+        forAll(arbitrary[String], Gen.option(arbitrary[String]), userAnswersGen.arbitrary) { (upeSafeId, fmSafeId, userAnswers) =>
+          val modifiedData = (userAnswers.data \ "upeGRSResponse" \ "partnershipEntityRegistrationData").as[JsObject] - "companyProfile"
```
Comment (JamesMMiller) [inline test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala:119]: Doing this rather than doing the quite verbose task of creating a whole new generator just for removing company profile

File: test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala (hunk 1 above)
Comment (ridow-hmrc) [reply test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala]: it's a style point. I'm happy for you to say you prefer it this way

**Review by ridow-hmrc — APPROVED** [review]
